src/typescript/frontend/src/app/dexscreener/pair.ts

```diff
@@ -0,0 +1,115 @@
+ //     "createdAtTxnId": "0xe9e91f1ee4b56c0df2e9f06c2b8c27c6076195a88a7b8537ba8313d80e6f124e",
+ //     "feeBps": 100
+ //   }
+ // }
+ **/
+
+import { type NextRequest, NextResponse } from "next/server";
+import {
+  fetchMarketRegistrationEventBySymbolBytes
+} from "@sdk/indexer-v2/queries/app/dexscreener";
+import { SYMBOL_EMOJI_DATA } from "@sdk/emoji_data";
+import { getAptosClient } from "@sdk/utils/aptos-client";
+
+
+/**
+ * - All `Pair` props are immutable - Indexer will not query a given pair more than once
+ * - In most cases, pair ids will correspond to contract addresses. Ids are case-sensitive.
+ * - `dexKey` is an identifier for the DEX that hosts this pair. For most adapters this will be a static value such as
+ * `uniswap`, but if multiple DEXes are tracked an id such as a factory address may be used.
+ * - `asset0` and `asset1` order should **never** change. `amount0/reserve0` will always refer to the same `asset0`,
+ * and `amount1/reserve1` will always refer to the same `asset1`. If asset order mutates then all data after the change
+ * will be invalid and a re-index will be required.
+ *     - A simple strategy to keep this in check is to simply order assets alphabetically. For example, in a pair
+ * containing assets `0xAAA` and `0xZZZ`, `asset0=0xAAA` and `asset1=0xZZZ`
+ *     - DEX Screener UI will automatically invert pairs as needed and default to their most logical order (i.e.:
+ * `BTC/USD` as opposed to `USD/BTC`)
+ * - `createdAtBlockNumber`, `createdAtBlockTimestamp` and `createdAtTxnId` are optional but encouraged. If unavailable
+ * DEX Screener can bet set to assume pair creation date is the same date as its first ever event.
+ * - `feeBps` corresponds to swap fees in bps. For instance, a fee of 1% maps to `feeBps=100`
+ * - `pool` is only recommended for DEXes that support multi-asset pools and allows the DEX Screener UI to correlate
+ * multiple pairs in the same multi-asset pool
+ * - `metadata` includes any optional auxiliary info not covered in the default schema and not required in most cases
+ */
+export interface Pair {
+  id: string;
+  dexKey: string;
+  asset0Id: string;
+  asset1Id: string;
+  createdAtBlockNumber?: number;
+  createdAtBlockTimestamp?: number;
+  createdAtTxnId?: string;
+  creator?: string;
+  feeBps?: number;
+  pool?: {
+    id: string;
+    name: string;
+    assetIds: string[];
+    pairIds: string[];
+    metadata?: Record<string, string>;
+  };
+  metadata?: Record<string, string>;
+}
+
+export interface PairResponse {
+  pair: Pair;
+}
+
+/**
+ *
+ * @param pairId is the pair ID. Generally it's `event.market.symbolEmojis.join("") + "-APT"`
+ */
+export async function getPair(pairId: string): Promise<Pair> {
+  const emojiString = pairId.split("-")[0];
+  const sybolEmojis = emojiString.split("");
+
+  const symbolBytesArray = sybolEmojis.map((e) => SYMBOL_EMOJI_DATA.byEmoji(e)!.bytes);
+  const symbolBytes = new Uint8Array(symbolBytesArray.flatMap((v) => Array.from(v)));
```

Comment:
```suggestion
  const symbolEmojis = getSymbolEmojisInString(emojiString);
  const symbolBytes = encodeEmojis(symbolEmojis);
```

Emojis are parsed really oddly; the zero-width joiners makes for a lot of confusing parsing behavior, because lots of emojis are a base emoji + some modifier to indicate the variant.

An obvious example is a person + a zwj + a skin tone. But there are others like the polar bear emoji, which is basically just `🐻\x200d❄️`, see more [here ](https://unicode.org/Public/emoji/16.0/emoji-zwj-sequences.txt) if you want, it's kind of interesting lol)

But yeah to reliably parse these use the above or I think some of these will get parsed incorrectly

Reply:
Yeah this was my fear as well- this is the main place I wanted to write some tests for, too, to ensure stability of the IDs acoss these endpoints- I'll take a read (it is interesting!) and use your helpers 🙏  thank you!